### DISCUSSION TITLE
Will the cities and products names change during the last run?

### DISCUSSION THREAD

**Discussion by webNeat** [geeksblabla/blanat#242] [Q&A] [answered]
I noticed that the top solutions (mine included) started to write custom hash-tables specifically for the cities and products on the `gen.py` script. This means that if a new or slightly different city/product is added on the input, these solutions may behave poorly or not work at all.

is it safe to assume that only the cities and products on the `gen.py` script will figure out in the final input?

**Comment by samir-** [comment, marked as answer]
Yes it was said in another thread I guess. That it's a fixed set and the file is not supposed to change
